src/main/java/teammates/common/util/Url.java

```diff
@@ -2,14 +2,12 @@
 
 public class Url {
 
+    protected final String baseUrl;
     protected String urlString;
 
-    public Url(String url) {
-        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
-    }
-
-    protected String getAppUrl() {
-        return Config.APP_URL;
+    public Url(String baseUrl, String url) {
```

Comment:
The last two seem like 'misuse cases'. Why can't they be like this (assuming we add a one parameter constructor)?
```java
new Url("http://www.google.com/page").getRelativePath() = "/page";
new Url("http://www.google.com/page").toString() = "http://www.google.com/page";
```
This is the only use case that can justify the current constructor, but I don't think we have a need for such usage.
```java
new Url("http://www.facebook.com", "http://www.google.com/page").toString() = "http://www.facebook.com/page";
```